fenix/app/src/main/java/org/mozilla/fenix/addons/ExtensionProcessDisabledController.kt

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.addons
+
+import android.app.AlertDialog
+import android.content.Context
+import androidx.annotation.UiContext
+import mozilla.components.browser.state.action.ExtensionProcessDisabledPopupAction
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.support.ktx.android.content.appName
+import mozilla.components.support.webextensions.ExtensionProcessDisabledPopupFeature
+import org.mozilla.fenix.GleanMetrics.Addons
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import org.mozilla.geckoview.WebExtensionController
+
+/**
+ * Present a dialog to the user notifying of extension process spawning disabled and also asking
+ * whether they would like to continue trying or disable extensions. If the user chooses to retry,
+ * enable the extension process spawning with [WebExtensionController.enableExtensionProcessSpawning].
+ *
+ * @param context to show the AlertDialog
+ * @param store The [BrowserStore] which holds the state for showing the dialog
+ * @param webExtensionController to call when a user enables the process spawning
+ * @param builder to use for creating the dialog which can be styled as needed
+ * @param appName to be added to the message. Necessary to be add as a param for testing
+ */
+private fun presentDialog(
+    @UiContext context: Context,
+    store: BrowserStore,
+    webExtensionController: WebExtensionController,
+    builder: AlertDialog.Builder,
+    appName: String,
+) {
+    val message = context.getString(R.string.addon_process_crash_dialog_message, appName)
+
+    builder.apply {
+        setCancelable(false)
+        setTitle(R.string.addon_process_crash_dialog_title)
+        setMessage(message)
+        setPositiveButton(R.string.addon_process_crash_dialog_retry_button_text) { dialog, _ ->
+            webExtensionController.enableExtensionProcessSpawning()
```

Comment:
Noticed we added this function to the engine now, should we replace the geckoView `webExtensionController` with an engine reference?